crates/oxc_linter/src/partial_loader/vue_partial_loader.rs

```diff
@@ -0,0 +1,496 @@
+            console.log("hi")
+            console.log("I am multi line")
+            console.log("<script></script>")
+            console.log(`<script></script>`)
+            console.log('<script></script>')
+--------"#
+        );
+    }
+
+    #[test]
+    fn test_build_vue_with_ts_flag_1() {
+        let source_text = r#"
+        <script lang="ts" setup>
+            1/1
+        </script>
+        "#;
+
+        let loader_value = VuePartialLoader::from(source_text).build();
+        assert!(loader_value.source_type.is_typescript());
+        assert_eq!(loader_value.source_text.trim(), "1/1");
+    }
+
+    #[test]
+    fn test_build_vue_with_ts_flag_2() {
+        let source_text = r"
+        <script lang=ts setup>
+            1/1
+        </script>
+        ";
+
+        let loader_value = VuePartialLoader::from(source_text).build();
+        assert!(loader_value.source_type.is_typescript());
+        assert_eq!(loader_value.source_text.trim(), "1/1");
+    }
+
+    #[test]
+    fn test_build_vue_with_ts_flag_3() {
+        let source_text = r"
+        <script lang='ts' setup>
+            1/1
+        </script>
+        ";
+
+        let loader_value = VuePartialLoader::from(source_text).build();
+        assert!(loader_value.source_type.is_typescript());
+        assert_eq!(loader_value.source_text.trim(), "1/1");
+    }
+
+    #[test]
+    fn test_build_vue_with_tsx_flag() {
+        let source_text = r"
+        <script lang=tsx setup>
+            1/1
+        </script>
+        ";
+
+        let loader_value = VuePartialLoader::from(source_text).build();
+        assert!(loader_value.source_type.is_jsx());
+        assert!(loader_value.source_type.is_typescript());
+        assert_eq!(loader_value.source_text.trim(), "1/1");
+    }
+
+    #[test]
+    fn test_build_vue_with_escape_string() {
+        let source_text = r"
+        <script setup>
+            a.replace(/&#39;/g, '\''))
+        </script>
+        <template> </template>
+        ";
+
+        let loader_value = VuePartialLoader::from(source_text).build();
+        assert!(!loader_value.source_type.is_typescript());
+        assert_eq!(loader_value.source_text.trim(), r"a.replace(/&#39;/g, '\''))");
+    }
+
+    #[test]
+    fn test_multi_level_template_literal() {
+        let source_text = r"
+        <script setup>
+            `a${b( `c \`${d}\``)}`
+        </script>
+        ";
+
+        let loader_value = VuePartialLoader::from(source_text).build();
+        assert_eq!(loader_value.source_text.trim(), r"`a${b( `c \`${d}\``)}`");
+    }
+
+    // TODO: fix this, current did know whether / is the start of regex or not
+    #[ignore]
+    #[test]
+    fn test_brace_with_regex_in_template_literal() {
+        let source_text = r"
+        <script setup>
+            `${/{/}`
+        </script>
+        ";
+
+        let loader_value = VuePartialLoader::from(source_text).build();
+        assert_eq!(loader_value.source_text.trim(), r"`${/{/}`");
+    }
```

Comment:
How I can solve it, I have no idea whether `/` is an operator or RegExp.

Reply:
You can try and parse it as an expression?